src/pgwire/src/protocol.rs

```diff
@@ -973,18 +985,30 @@ where
             {
                 Some(portal) => portal,
                 None => {
+                    let msg = format!("portal {} does not exist", portal_name.quoted());
+                    if let Some(outer_ctx_extra) = outer_ctx_extra {
+                        self.adapter_client.retire_execute(
+                            outer_ctx_extra,
+                            StatementEndedExecutionReason::Errored { error: msg.clone() },
+                        );
+                    }
                     return self
-                        .error(ErrorResponse::error(
-                            SqlState::INVALID_CURSOR_NAME,
-                            format!("portal {} does not exist", portal_name.quoted()),
-                        ))
+                        .error(ErrorResponse::error(SqlState::INVALID_CURSOR_NAME, msg))
```

Comment:
`self.error` is called in a buncha places. Can we pass it `outer_ctx_extra` so it can do the retire logic? Are there any calls to `error` that don't want that to happen?

Reply:
It would be a bit annoying because we call that from tons of places that aren't in the context of a statement execution. So we'd have to pass `None` in all those places.